check-stylish.sh

```diff
@@ -0,0 +1,46 @@
+#!/bin/bash
+
+set -e
+
+function fail_maybe_explain {
+    if [ -n "$TRAVIS" ];
+    then
+        echo "Run ./check-stylish.sh locally to stylize your files / debug this"
+    fi
+    exit 1
+}
+
+if [ -n "$TRAVIS" ];
+then
+    # It takes a while to compile, let's not tax Travis' infrastructure for no
+    # reason. We can't cache it via cabsnap because that's based on the install-
+    # plan for Idris, which of course doesn't have stylish-haskell as a
+    # dependency.
+    STYLISH_EXE=$(mktemp)
+    wget -O $STYLISH_EXE "https://s3-us-west-2.amazonaws.com/idris-travis-binaries/stylish-haskell"
```

Comment:
I don't understand the reference to `cabsnap`, and DuckDuckGoing it didn't return anything I could understand as related. Is that a name for Travis's built-in caching? Why can't we keep the binary in the Travis cache instead of hosting it on an AWS site that we don't really control? I worry that it might disappear.

If we have to host a binary, can we host it on idris-lang.org instead?